bin/render-activity-stream-html.js

```diff
@@ -100,26 +258,101 @@ function main() {
+  // Save default locale's strings to compare against other locales' strings
+  let defaultStrings;
+  const hasOnlyDefaultStrings = strings => defaultStrings &&
+    !Object.keys(strings).some(key => strings[key] !== defaultStrings[key]);
+
+  // Process the default locale first then all the ones from mozilla-central
+  const localizedLocales = [];
+  for (const locale of [DEFAULT_LOCALE, ...MOZILLA_CENTRAL_LOCALES]) {
+    const strings = getStrings(locale, allStrings);
+    if (hasOnlyDefaultStrings(strings)) {
+      console.log(`Skipping "${locale}" because it uses the same strings as ${DEFAULT_LOCALE}.`);
+      continue;
+    }
+
+    const {html, state} = prerender(locale, strings);
+    const options = Object.assign({}, baseOptions, {
+      direction: state.App.textDirection,
+      locale,
+      title: state.App.strings.newtab_page_title
+    });
+
+    // Put locale-specific files in their own directory
+    const basePath = path.join(addonPath, "prerendered", locale);
+    mkdir("-p", basePath);
+    for (const [file, templater] of localizedFiles) {
+      fs.writeFileSync(path.join(basePath, file), templater({html, options, state}));
+    }
+    console.log(`Done writing "${locale}" localized files to ${basePath}.`);
+
+    // When handling multiple locales, do the following once (with the default)
+    if (locale === DEFAULT_LOCALE) {
+      fs.writeFileSync(path.join(basePath, "activity-stream-debug.html"),
+        templateHTML(Object.assign({}, options, {debug: true})));
+      fs.writeFileSync(path.join(basePath, "activity-stream-prerendered-debug.html"),
+        templateHTML(Object.assign({}, options, {debug: true}), html));
+      console.log(`Done writing debug html to ${basePath}.`);
+
+      // Save the default locale's strings to compare against other locales' strings
+      defaultStrings = strings;
+    }
+
+    localizedLocales.push(locale);
+  }
+
+  // Prepare lines that we'll need for updating jar.mn
+  const defineLines = [];
+  const ifLines = [];
+  const elseLines = ["#else"];
+  for (const locale of localizedLocales) {
+    let targetLines;
+    if (locale === DEFAULT_LOCALE) {
+      targetLines = elseLines;
+    } else {
+      // Preprocessor can't do "-" in conditions (SYNTAX_ERR) so indirection!
+      const condLocale = locale.replace(/-/g, "_");
+      if (condLocale !== locale) {
+        defineLines.push(`#define ${condLocale} ${locale}`);
+      }
+      ifLines.push(`#${ifLines.length ? "elif" : "if"} AB_CD == ${condLocale}`);
+      targetLines = ifLines;
+    }
+    for (const file of localizedFiles.keys()) {
+      targetLines.push(`  content/data/content/${file} (./prerendered/${locale}/${file})`);
+    }
+  }
 
-  console.log(`Done writing html and js to ${BASE_FILE_PATH}.`); // eslint-disable-line no-console
+  // Replace existing render-generated lines in jar.mn
+  const scriptName = path.basename(__filename);
```

Comment:
Can you actually include a complete path here if someone is trying to find this when looking at mc?

Reply:
What's a "complete" path? "bin/render-activity-stream-html.js" or "https://github.com/mozilla/activity-stream/blob/master/bin/render-activity-stream-html.js"

Reply:
Used full url:
`# These defines and ifs below are generated by https://github.com/mozilla/activity-stream/blob/master/bin/render-activity-stream-html.js`